native/src/components/MapView.tsx

```diff
@@ -82,38 +78,23 @@ const MapView = ({
     bounds: coordinates ? undefined : bounds
   }
 
-  const onLocationPermissionRequest = useCallback((locationPermission: PermissionStatus | undefined) => {
-    const permissionGranted = locationPermission === RESULTS.GRANTED
-    setLocationPermissionGranted(permissionGranted)
-  }, [])
-
-  useEffect(() => {
-    checkLocationPermission().then(onLocationPermissionRequest)
-  }, [onLocationPermissionRequest])
-
-  const requestPermission = useCallback(() => {
-    requestLocationPermission().then(onLocationPermissionRequest)
-    locationPermissionGranted && setFollowUserLocation(!followUserLocation)
-  }, [followUserLocation, locationPermissionGranted, onLocationPermissionRequest])
+  const onRequestLocation = useCallback(() => {
+    onRequestLocationPermission()
+    setFollowUserLocation(locationPermissionGranted)
```

Comment:
Does this actually work? The prop `locationPermissionGranted` is not updated immediately, is it?

Reply:
updated, as onRequestLocationPermission actually returns a promise